x-pack/plugins/security_solution/common/endpoint/types.ts

```diff
@@ -76,12 +76,18 @@ export interface ResolverNodeStats {
  */
 export interface ResolverChildNode extends ResolverLifecycleNode {
   /**
-   * A child node's pagination cursor can be null for a couple reasons:
-   * 1. At the time of querying it could have no children in ES, in which case it will be marked as
-   *  null because we know it does not have children during this query.
-   * 2. If the max level was reached we do not know if this node has children or not so we'll mark it as null
+   * nextChild can have 3 different states:
```

Comment:
Just a thought, I understand what this is saying, especially after reading through your write up, but one of @oatkiller's famous diagrams may be useful here, you could probs just copy paste one he's already done. Not necessary, but I think it could be helpful

Reply:
Thanks @michaelolo24 do you mean an ascii diagram of a tree? Or a table?

Reply:
ascii I think

Reply:
I think what I'll probably do is put a link to the relevant section in the docs that I'll be adding as a follow up or add on to this PR.